extensions/jaeger/runtime/src/main/java/io/quarkus/jaeger/runtime/JaegerDeploymentRecorder.java

```diff
@@ -1,45 +1,37 @@
 package io.quarkus.jaeger.runtime;
 
-import static io.jaegertracing.Configuration.JAEGER_SERVICE_NAME;
-
 import java.util.Optional;
 import java.util.function.Function;
 
-import org.eclipse.microprofile.config.Config;
-import org.eclipse.microprofile.config.ConfigProvider;
 import org.jboss.logging.Logger;
 
 import io.opentracing.util.GlobalTracer;
+import io.quarkus.runtime.ApplicationConfig;
 import io.quarkus.runtime.annotations.Recorder;
 
 @Recorder
 public class JaegerDeploymentRecorder {
-    private static volatile boolean registered;
-
     private static final Logger log = Logger.getLogger(JaegerDeploymentRecorder.class);
+    private static final Optional UNKNOWN_SERVICE_NAME = Optional.of("quarkus/unknown");
+    private static final QuarkusJaegerTracer quarkusTracer = new QuarkusJaegerTracer();
 
-    public void registerTracer(JaegerConfig jaeger) {
-        if (!registered) {
-            if (isValidConfig(jaeger)) {
-                initTracerConfig(jaeger);
-                QuarkusJaegerTracer quarkusJaegerTracer = new QuarkusJaegerTracer();
-                log.debugf("Registering tracer to GlobalTracer %s", quarkusJaegerTracer);
-                GlobalTracer.register(quarkusJaegerTracer);
-            }
-            registered = true;
+    static {
```

Comment:
TBH this looks a little suspicious. Can you please explain what you are trying to achieve here?

Reply:
`GlobalTracer` is used to pass the tracer to CDI producer defined in MP-OpenTracing spec.

This block registers the quarkus tracer to the global tracer. GlobalTracer has is a static accessor to a registered tracer, and it does not have a reset method - hence I have added it to static block. 

I have moved it to `registerTracer`  method - It might be better if a caller resets the tracer (via reflection - there is a test-jar utility).